sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -127,14 +127,18 @@
     Mono<Response<AreaCodes>> getAllAreaCodesWithResponse(
         String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions,
         Context context) {
-        Objects.requireNonNull(locationType, "'locationType' cannot be null.");
-        Objects.requireNonNull(countryCode, "'countryCode' cannot be null.");
-        Objects.requireNonNull(phonePlanId, "'phonePlanId' cannot be null.");
-
-        LocationOptionsQueries locationOptionsQueries = new LocationOptionsQueries();
-        locationOptionsQueries.setLocationOptions(locationOptions);
-
         try {
+            if (locationOptions == null) {
+                Mono.error(new NullPointerException("'locationType' cannot be null."));
+            }
+            if (countryCode == null) {
+                Mono.error(new NullPointerException("'countryCode' cannot be null."));
+            }
+            if (phonePlanId == null) {
+                Mono.error(new NullPointerException("'phonePlanId' cannot be null."));
+            }
```

Comment:
You don't need to put Mono.error inside the try block.. simply have the Objects.requireNonNull statements here and if they throw they will be caught and returned as Mono errors